frontend/src/components/pages/rp-connect/secrets/Secrets.Create.tsx

```diff
@@ -0,0 +1,142 @@
+import {formatPipelineError} from '../errors';
+import {CreateSecretRequest, Scope} from '../../../../protogen/redpanda/api/dataplane/v1alpha2/secret_pb';
+
+const {ToastContainer, toast} = createStandaloneToast();
+
+const returnSecretTab = '/connect-clusters?defaultTab=redpanda-connect-secret'
+
+
+@observer
+class RpConnectSecretCreate extends PageComponent {
+
+    @observable id = '';
+    @observable secret = '';
+    @observable isCreating = false;
+
+    constructor(p: any) {
+        super(p);
+        makeObservable(this, undefined, {autoBind: true});
+    }
+
+    initPage(p: PageInitHelper) {
+        p.title = 'Create Secret';
+        p.addBreadcrumb('Redpanda Connect Secret Manager', '/rp-connect/secret/create');
+        p.addBreadcrumb('Create Secret', '');
+
+        this.refreshData(true);
+        appGlobal.onRefresh = () => this.refreshData(true);
+    }
+
+    refreshData(_force: boolean) {
+        rpcnSecretManagerApi.refreshSecrets(_force);
+    }
+
+    cancel() {
+        this.secret = '';
+        this.id = '';
+        appGlobal.history.push(returnSecretTab);
+    }
+
+    async createSecret() {
+        this.isCreating = true;
+
+        //create function given string return base64 encoded Uint8Array
+        function base64Encode(str: string): Uint8Array {
+            const encodedString = btoa(str);
+            const charList = encodedString.split('').map(char => char.charCodeAt(0));
+            return new Uint8Array(charList);
+        }
+
+
+        rpcnSecretManagerApi.create(new CreateSecretRequest({
+            id: this.id,
+            secretData: base64Encode(this.secret),
+            scopes: [Scope.REDPANDA_CONNECT]
+        }))
+            .then(async () => {
+                toast({
+                    status: 'success', duration: 4000, isClosable: false,
+                    title: 'Secret created'
+                });
+                await pipelinesApi.refreshPipelines(true);
+                appGlobal.history.push(returnSecretTab);
+            })
+            .catch(err => {
+                toast({
+                    status: 'error', duration: null, isClosable: true,
+                    title: 'Failed to create secret',
+                    description: formatPipelineError(err),
+                })
+            })
+            .finally(() => {
+                this.isCreating = false;
+            });
+    }
+
+    render() {
+        if (!rpcnSecretManagerApi.secrets) return DefaultSkeleton;
+
+        const alreadyExists = (rpcnSecretManagerApi.secrets || []).any(x => x.id == this.id);
+        const isIdEmpty = this.id.trim().length == 0;
+        const isSecretEmpty = this.secret.trim().length == 0;
+
+        return (
+            <PageContent>
+                <ToastContainer/>
+                <Flex flexDirection="column" gap={5}>
+                    <FormField label="Secret name" isInvalid={alreadyExists} errorText="Secret name is already in use">
```

Comment:
Is that always going to be the reason for the error? What if some invalid character is used? Wouldn't it show wrong error text?